packages/agreement-process/src/services/agreementStateProcessor.ts

```diff
@@ -144,3 +170,181 @@ export const suspendedByProducerFlag = (
+  consumer: CompactTenant,
+  eservices: EService[],
+  correlationId: string,
+  logger: Logger
+): Promise<CreateEvent<AgreementEvent> | void> {
+  const descriptor = eservices
+    .find((eservice) => eservice.id === agreement.data.eserviceId)
+    ?.descriptors.find(
+      (descriptor) => descriptor.id === agreement.data.descriptorId
+    );
+
+  if (!descriptor) {
+    logger.error(
+      `Descriptor ${agreement.data.descriptorId} not found for Agreement ${agreement.data.id} - EService ${agreement.data.eserviceId}`
+    );
+    return;
+  }
+
+  const nextState = nextStateByAttributes(agreement.data, descriptor, consumer);
+
+  const newSuspendedByPlatform = suspendedByPlatformFlag(nextState);
+
+  const finalState = agreementStateByFlags(
+    nextState,
+    agreement.data.suspendedByProducer,
+    agreement.data.suspendedByConsumer,
+    newSuspendedByPlatform
+  );
+
+  const updatedAgreement: Agreement = {
+    ...agreement.data,
+    state: finalState,
+    suspendedByPlatform: newSuspendedByPlatform,
+  };
+
+  if (allowedStateTransitions(agreement.data.state).includes(finalState)) {
+    if (newSuspendedByPlatform !== agreement.data.suspendedByPlatform) {
+      return match(finalState)
+        .with(agreementState.suspended, () =>
+          toCreateEventAgreementSuspendedByPlatform(
+            updatedAgreement,
+            agreement.metadata.version,
+            correlationId
+          )
+        )
+        .with(agreementState.active, () =>
+          toCreateEventAgreementUnsuspendedByPlatform(
+            updatedAgreement,
+            agreement.metadata.version,
+            correlationId
+          )
+        )
+        .with(agreementState.missingCertifiedAttributes, () =>
+          toCreateEventAgreementPutInMissingCertifiedAttributesByPlatform(
+            updatedAgreement,
+            agreement.metadata.version,
+            correlationId
+          )
+        )
+        .with(agreementState.draft, () =>
+          toCreateEventAgreementPutInDraftByPlatform(
+            updatedAgreement,
+            agreement.metadata.version,
+            correlationId
+          )
+        )
+        .otherwise(
+          () =>
+            void logger.error(
+              `Agreement state transition not allowed from ${agreement.data.state} to ${finalState} - Agreement ${agreement.data.id} - EService ${agreement.data.eserviceId} - Consumer ${consumer.id}`
+            )
+        );
+    } else {
+      logger.error(
+        `Skipping Agreement state transition from ${agreement.data.state} to ${finalState}, no change in suspendedByPlatform flag - Agreement ${agreement.data.id} - EService ${agreement.data.eserviceId} - Consumer ${consumer.id}`
+      );
+    }
+  } else {
+    logger.error(
+      `Agreement state transition not allowed from ${agreement.data.state} to ${finalState} - Agreement ${agreement.data.id} - EService ${agreement.data.eserviceId} - Consumer ${consumer.id}`
+    );
```

Comment:
this is not an error and it does need to be logged.
This happens when there are transitions not useful fo this logic
For example:
- when the state of an agreement is not changed
- when the result of the FSM is `PENDING` from a `DRAFT` agreement. This is allowed as a transition, but not in this context (it can be done only when the consumer submit the agreement, not when the platform calculates the updates)
- etc

Maybe `allowed` in the name of the variable is a bit strong and this caused the confusion